src/Extensions/LeftAndMainSubsites.php

```diff
@@ -45,6 +47,42 @@ protected function onInit()
         Requirements::javascript('silverstripe/subsites:client/dist/js/LeftAndMain_Subsites.js');
     }
 
+    public static function get_template_global_variables()
+    {
+        return [
+            'SubsiteSwitchList',
+        ];
+    }
+
+    /**
+     * Generates a list of subsites with the data needed to
+     * produce a dropdown site switcher
+     * @return SS_List<Subsite>
+     */
+    public static function SubsiteSwitchList(): SS_List
```

Comment:
Since `SubsiteXHRController` isn't a subclass of `LeftAndMain` anymore, it doesn't get access to the methods in this extension. This method is only needed in templates anyway so I've just made it _always_ available in templates.

Since I had to break the method signature to make it static I took the liberty of renaming it to make it much clearer what its purpose is.

The alternative would be to make a separate extension with just this logic and apply it to both `LeftAndMain` and the xhr controller. Happy to do that if you prefer that approach.